### ISSUE TITLE
fixed-hash: cannot define Hash struct, like 'H256', in constants directly

### ISSUE THREAD

**Issue by koushiro** [paritytech/parity-common#109]
fixed-hash v0.3.0 cannot define Hash struct in constants directly, like:

```rust
pub const TEST_HASH: H256 = H256([0xc5, 0xd2, 0x46, 0x01, 0x86, 0xf7, 0x23, 0x3c, 0x92, 0x7e, 0x7d, 0xb2, 0xdc, 0xc7, 0x03, 0xc0, 0xe5, 0x00, 0xb6, 0x53, 0xca, 0x82, 0x27, 0x3b, 0x7b, 0xfa, 0xd8, 0x04, 0x5d, 0x85, 0xa4, 0x70]);
```

In version 0.2.5, the definition of Hash structures is
```rust
pub struct $from (pub [u8; $size]);
```
but in version 0.3.0, the definition of Hash structure is 
```rust
$visibility struct $name ([u8; $n_bytes]);
```

Why does the new version of `fixed-hash` change the public field in the Hash structure to the private one?

**Comment by Robbepop** [comment]
It was changed to have a better encapsulated state so that immutable references to hashes cannot change them. Also with Rust's progress on `const_fn` we can expect this to work in const context in the future. For now it doesn't. I myself am open to revert this again for the sake of usability.

**Comment by koushiro** [comment]
@Robbepop Thanks for your reply! Maybe I can submit a PR to revert this and upgrade `fixed-hash` to v0.3.1?
